packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,136 @@
+    square: { type: Boolean as PropType<boolean>, default: false },
+    shadow: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: false },
+    icon: { type: String as PropType<string>, default: '' },
+    tag: { type: String as PropType<string>, default: 'span' },
+
+    size: {
+      type: String as PropType<string>,
+      default: 'medium',
+      validator: (value: string) => {
+        return ['small', 'medium', 'large'].includes(value)
+      },
+    },
+  },
+
+  setup (props, { emit }) {
     const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
 
-    return {
-      hasKeyboardFocus,
-      keyboardFocusListeners,
-    }
-  })
+    const { getColor } = useColors()
+    const colorComputed = computed(() => getColor(props.color))
+    const borderColor = computed(() => props.outline ? colorComputed : '')
 
-  hoverState = false
-  focusState = false
+    const { tagComputed, hrefComputed } = useRouterLink(props)
 
-  created () {
-    watch(() => this.hoverState, (value) => {
-      this.updateHoverState(value)
-    })
-  }
+    const hoverState = ref(false)
+    const updateHoverState = (isHover: boolean) => {
+      hoverState.value = isHover
+    }
 
-  get iconSize () {
-    const size: any = {
+    const size = {
       small: '0.875rem',
       medium: '1rem',
       large: '1.25rem',
-    }
-    return size[this.size]
-  }
+    } as Record<string, string>
+    const iconSize = computed(() => size[props.size])
 
-  get indexComputed () {
-    return this.disabled ? -1 : 0
-  }
+    const indexComputed = computed(() => props.disabled ? -1 : 0)
 
-  get computedClass () {
-    return {
-      'va-chip--small': this.size === 'small',
-      'va-chip--large': this.size === 'large',
-      'va-chip--square': this.square,
-      'va-chip--disabled': this.disabled,
-    }
-  }
+    const computedClass = computed(() => ({
+      'va-chip--small': props.size === 'small',
+      'va-chip--large': props.size === 'large',
+      'va-chip--square': props.square,
+      'va-chip--disabled': props.disabled,
+    }))
 
-  get shadowStyle () {
-    if (!this.shadow || this.flat || this.outline || this.disabled || this.SetupContext.hasKeyboardFocus) {
-      return
-    }
-    return '0 0.125rem 0.19rem 0 ' + getBoxShadowColor(this.colorComputed)
-  }
+    const shadowStyle = computed(() => {
+      if (!props.shadow || props.flat || props.outline || props.disabled || hasKeyboardFocus.value) {
+        return
+      }
+      return `0 0.125rem 0.19rem 0 ${getBoxShadowColor(colorComputed.value)}`
+    })
 
-  get computedStyle () {
-    const borderColor = this.outline ? this.colorComputed : ''
-    const computedStyle = {
-      color: this.colorComputed,
-      borderColor,
-      background: '',
-      boxShadow: this.shadowStyle,
-    }
+    const computedStyle = computed(() => {
+      const result = reactive({
```

Comment:
There is no reason to create reactive variable inside computed.

Reply:
Without reactive `borderColor` in not working in proper way. Think just can move it in `computedStyle`.